sdk/core/azure-core/inc/http/http.hpp

```diff
@@ -3,27 +3,89 @@
 
 #pragma once
 
+#include <string>
+#include <vector>
+
 #include <internal/contract.hpp>
 
+using namespace std;
+
 namespace azure
 {
 namespace core
 {
 namespace http
 {
 
-class http_request
+namespace http_method
 {
+enum HttpMethod
+{
+  GET,
+  HEAD,
+  POST,
+  PUT,
+  DELETE,
+  PATCH,
+};
+}
 
+class Header
+{
 private:
-  /* data */
+  string name;
+  string value;
+
 public:
-  http_request();
-  ~http_request();
+  Header(string name, string value)
+  {
+    this->name = name;
+    this->value = value;
+  }
+  ~Header() {}
+  string getName();
+  string getValue();
 };
 
-http_request::http_request() {}
-http_request::~http_request() {}
+class Request
+{
+
+private:
+  http_method::HttpMethod method;
+  string url;
+  std::vector<Header> headers;
+  string body;
+  int32_t query_start; // 0 = no query in url, > 0 = query start position in url '?'
+
+  inline int32_t get_query_start(string url)
+  {
+    auto position = url.find('?');
+    return position == string::npos ? 0 : position;
+  }
+
+public:
+  Request(http_method::HttpMethod httpMethod, string url)
+  {
+    this->method = httpMethod;
+    this->url = url;
+    this->query_start = get_query_start(url);
+  }
+
+  Request(http_method::HttpMethod httpMethod, string url, string body) : Request(httpMethod, url)
```

Comment:
std::string const& url, std::string const& body

Reply:
and so on. Everywhere where you take a string parameter, take it as a constant reference.